locator/topology.hh

```diff
@@ -264,6 +264,12 @@ public:
         return _dc_endpoints;
     }
 
+    const std::unordered_map<sstring,
+                            std::unordered_set<const node*>>&
+    get_datacenter_nodes() const {
+        return _dc_nodes;
+    }
+
     const std::unordered_map<sstring,
```

Comment:
@bhalevy we should "compile" the datacenter names to indexes when we build token_metadata etc.

Reply:
or even: pointers to datacenter objects

Reply:
Yes. We can probably use datacenter* within a topology instance similar to the way we use node* and introduce a level of indirection in endpoint_dc_rack to get the the dc and rack names via pointers to the objects (that would contain their named and members)